scripts/create-hashes/index.ts

```diff
@@ -0,0 +1,70 @@
+import readline from "node:readline";
+import { createHash } from "node:crypto";
+import { readFileSync } from "node:fs";
+
+interface InputLine {
+  location?: string;
+  base?: string;
+}
+
+interface HashRecord {
+  base?: string;
+  file?: string;
+  sha384?: string;
+  sha512?: string;
+}
+
+main().catch((error) => {
+  console.error(error);
+
+  process.exit(1);
+});
+
+function createFileHash(algorithm: string, filename: string, base: string) {
+  const file_buffer = readFileSync(`${base}/${filename}`);
+  const hash = createHash(algorithm);
+
+  hash.update(file_buffer);
+  return btoa(hash.digest("binary"));
+}
+
+function includeFile(filename: string) {
+  const ignoreThese = [".scss", "readme.md", "README.md", "CHANGELOG.md", "DISCLAIMER.txt", ".d.ts"];
+
+  let result = true;
+  for (const ignore of ignoreThese) {
+    if (filename.endsWith(ignore)) {
+      result = false;
+      break;
+    }
+  }
+
+  return result;
```

Comment:
```suggestion
  return ignoreThese.every((exclude) => !filename.endsWith(exclude));
  
  // of
  // return !ignoreThese.some((exclude) => filename.endsWith(exclude));
```

Maar mooier is een package zoals [`minimatch`](https://www.npmjs.com/package/minimatch). Want we gaan ongetwijfeld een keertje nat op iets wat een dedicated package heeft opgelost.

En het is ook duidelijker: Je hebt 'n lijst van array's maar pas verderop in de code zie je dat er met `String.prototype.endsWith()` wordt getest. Een glob is alomvattend.

It's up to you

Reply:
Kwam er niet direct uit met `minimatch`. Wel de suggestie met `.every(` overgenomen.